### ISSUE TITLE
Unclear documentation on how to get prow tackle command

### ISSUE THREAD

**Issue by lioramilbaum** [kubernetes/test-infra#11352]
The documentation states one needs the command to start. 
Nothing is specified as regards to how to get it.

**Comment by stevekuznetsov** [comment]
/cc @cjwagner 

@lioramilbaum Hi! Could you explain which part of the documentation led you to needing `tackle` and what you were trying to do? It's not an often-used command.

We need to update the docs to just instruct you to `bazel run //prow/cmd/tackle -- --opts`

**Comment by lioramilbaum** [comment]
I would like to deploy Prow in a TEST environment. I am starting from scratch. The documentation leads me to using tackle.
If this is not the default deploy method and using bazel is the preferred one, than one should start with installing bazel. Right?

**Comment by stevekuznetsov** [comment]
How much of a test environment are you trying to create? I don't know if we really support a test prow well right now. Bazel is not a deployment method, I thought you were trying to run `tackle` locally.

**Comment by lioramilbaum** [comment]
I am trying to figure out how Prow works. Deploying it in a TEST environment is how I plan to investigate it.

**Comment by stevekuznetsov** [comment]
I am not certain that this will be very fruitful for you, you could try actually setting up webhooks but running in dry-run mode. @cjwagner WDYT is the best way forward here?

**Comment by cjwagner** [comment]
Typically if you want to deploy a Prow instance for demo/testing/educational purposes you just deploy Prow and configure it to operate on a dummy personal repo. 
See the instructions here for using `tackle`: https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#tackle-deployment

```
# Ideally use https://bazel.build, alternatively try:
#   go get -u k8s.io/test-infra/prow/cmd/tackle && tackle
bazel run //prow/cmd/tackle
```
`tackle` is just a go program that helps deploy Prow. Bazel is a build system that can be used to build and run it, but using `go get` or `go build` also works. Tackle is a relatively new utility so if you encounter problems with it please open an issue!
You can also follow the manual deployment steps here: https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#manual-deployment (`tackle` just automates these steps)

**Comment by lioramilbaum** [comment]
@cjwagner the page you referred to is the one which is unclear.
The first commented line points out what is the ideal approach with no clear instructions.
The second commented line didn't work for me till I figure it out. The GOPATH was not set.
Would you prefer I edit the page and add detailed instructions according to my experience?

**Comment by cjwagner** [comment]
> Would you prefer I edit the page and add detailed instructions according to my experience?

That would be great! I'm not quite sure what is unclear about the first line, but if you can improve it I'm all for it. I don't think embedding the installation instructions for bazel in our docs is appropriate though if that is what you are suggesting.
Similarly a properly configured GOPATH is required for using most (all?) features of go so I don't think we should be including details about that for this specific command.

It sounds like it might be most appropriate to add a `Prerequisites` section near the top of the getting_started_deploy.md guide. We can explicitly mention that go needs to be installed and configured (https://golang.org/doc/install) and also bazel if you want to build images or use bazel rules to deploy instead of applying k8s resource yaml directly.
WDYT?

**Comment by lioramilbaum** [comment]
Agree

**Comment by yb172** [comment]
Hello.
I've faced the same issue. I was reading about prow and wanted to try to deploy and try it. I created GKE cluster and found [Deploying prow](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md) page.

tl;dr:
 * Having "working `bazel`" in "prerequisites" would have helped me (link to bazel installation would be also nice)
 * Specifying that commands should be executed from `test-infra` would have also helped me

Longer version: here is my experience interacting with this page:
 * Ok, I'll need 3 things
 * Prow `tackle` command - don't know what it is but I see section below with `go get` so I guess I'll just get this util
 * Personal access token - ok, there is a link that leads to github personal access token generation page. Let's create token. Instruction says `Bot user must have write access to the repo`. Hmm... Am I supposed to create a bot? And where do I set "write access" to the repo? Ok, I guess just first section should be enough - rest is about teams and other things which don't seem relevant
 * Going back to doc: ah, here we go: `public_repo` and `repo:status` scopes. Ok, let's set that and generate token
 * Ok, next part is to run `tackle` and there is `go get` instruction - I've written my services in go and I know what `go get` does so I'll use `go get` version of `tackle`.
 * I'm in my project directory, run `go get -u k8s.io/test-infra/prow/cmd/tackle && tackle`
 * Ok, it's asking which context to use. Nice! That would be GKE that I've set up a few minutes ago.
 * Something about config and upstream. I guess enter would mean default... Hit enter - yes it is. kube resources installation is in progress
 * Ok, path to token: `~/.github/token` - failed, "no such file or directory". Hmm... Let me try again, maybe I made a typo
 * Run `tackle` one more time - this time it asks if I want to overwrite prow? no - it exited (well, what I've expected?)
 * Run `tackle` one more time and this time type `yes` - resources "unchanged", asks token. Verify path with `less ~/.github/token`, copy-paste - still failing.
 * Ok, maybe I should use absolute path?
 * Trying one more time with absolute path - it works! Created secrets, got ingress URL...
 * Asking about org or org/repo. Provided
 * **Error: `FATA[0053] Could not configure repos to send http://<ip>/hook webhooks.  error="find http://<ip>/hook hook in yb172: list hooks: return code not 2XX: 404 Not Found"` Hmm... What is this ip? Is it related to bot account?**
 * **After some time tried `tackle` one more time - and it worked!** `Enjoy your prow instance`. Great.
 * Ok, what's next - create `plugins.yaml` and `config.yaml`. Should I place these in repo root or in project directory? Let's try project directory.
 * Next step is to verify config - and there is no `go get` command so I should run `bazel`.
 * **Run verification command: error: `ERROR: The 'run' command is only supported from within a workspace.`** Well, I guess this command runs bazel target and I'm currently in my project repository which doesn't use bazel (even if it did there would be no `//prow/cmd/checkconfig`.
 * So I guess **I have to clone `test-infra` and run verification command from there**.
 * In order to do that I have to have bazel installed - let's do that
 * Ok, bazel is there, I'm in `test-infra`, let's run the command with path to my project - success!

Sorry about rather long description but I hope it would provide some clarity to where it was confusing.

**Comment by stevekuznetsov** [comment]
@yb172 your detailed feedback is really helpful, thank you for sharing that. We will try to update the docs to be better, but also if you want to take a stab at it and make the parts of the docs you found confusing better, I would be grateful!

**Comment by yb172** [comment]
@stevekuznetsov I'd be happy to but I thought @lioramilbaum is already on it. @lioramilbaum please let me know if you're still working on updating docs. If not I could take it over.

**Comment by lioramilbaum** [comment]
@yb172 didn't have the time to get into it yet. go ahead and take it over.
